File: datamodel/src/main/java/datamodel/clinical/ECG.java
package datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ECG {

    public abstract boolean hasSigAberrationLatestECG();

    @Nullable
    public abstract String aberrationDescription();

    @Nullable
    public abstract ECGMeasure qtcfMeasure();

    @Nullable
    public abstract ECGMeasure jtcMeasure();
}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/ECG.kt
package com.hartwig.actin.clinical.datamodel

data class ECG(
    val hasSigAberrationLatestECG: Boolean,
    val aberrationDescription: String?,
    val qtcfMeasure: ECGMeasure?,
    val jtcMeasure: ECGMeasure?
)
